main.go

```diff
@@ -1,167 +1,97 @@
 package main
 
 import (
-	"encoding/json"
 	"errors"
 	"fmt"
-	"io/ioutil"
-	"log"
-	"os"
-	"path"
+	"github.com/sensu/sensu-go/types"
+	"github.com/sensu/sensu-plugins-go-library/sensu"
 	"strings"
 
 	"github.com/bluele/slack"
-	"github.com/sensu/sensu-go/types"
-	"github.com/spf13/cobra"
 )
 
-type HandlerConfigOption struct {
-	Value string
-	Path  string
-	Env   string
-}
-
 type HandlerConfig struct {
-	SlackWebhookUrl HandlerConfigOption
-	SlackChannel    HandlerConfigOption
-	SlackUsername   HandlerConfigOption
-	SlackIconUrl    HandlerConfigOption
-	Timeout         int
-	Keyspace        string
+	sensu.PluginConfig
+	SlackWebhookUrl string
+	SlackChannel    string
+	SlackUsername   string
+	SlackIconUrl    string
 }
 
 var (
-	stdin  *os.File
 	config = HandlerConfig{
-		// default values
-		SlackWebhookUrl: HandlerConfigOption{Path: "webhook-url", Env: "SENSU_SLACK_WEHBOOK_URL"},
-		SlackChannel:    HandlerConfigOption{Path: "channel", Env: "SENSU_SLACK_CHANNEL"},
-		SlackUsername:   HandlerConfigOption{Path: "username", Env: "SENSU_SLACK_USERNAME"},
-		SlackIconUrl:    HandlerConfigOption{Path: "icon-url", Env: "SENSU_SLACK_ICON_URL"},
-		Timeout:         10,
-		Keyspace:        "sensu.io/plugins/slack/config",
+		PluginConfig: sensu.PluginConfig{
+			Name:     "sensu-slack-handler",
+			Short:    "The Sensu Go Slack handler for notifying a channel",
+			Timeout:  10,
+			Keyspace: "sensu.io/plugins/slack/config",
+		},
 	}
-	options = []*HandlerConfigOption{
-		// iterable slice of user-overridable configuration options
-		&config.SlackWebhookUrl,
-		&config.SlackChannel,
-		&config.SlackUsername,
-		&config.SlackIconUrl,
+
+	slackConfigOptions = []*sensu.PluginConfigOption{
+		{
+			Path:      "webhook-url",
```

Comment:
Since the `Path` and `Argument` fields for all of these options are the same, I would consider defining constants for them.